src/main/java/http/HttpRequestParser.java

```diff
@@ -0,0 +1,15 @@
+package http;
+
+public class HttpRequestParser {
+    private HttpRequestParser() {}
+
+    public static HttpRequest parse(final String rawRequestStr) {
```

Comment:
HttpRequestParser 의 등장으로 불필요한 단계가 추가되었다는 느낌을 받네요
아래와 같이 변경해서 HttpRequest 로 메서드를 옮겨보면 어떨까 생각해요

```suggestion
public static HttpRequest parse(final String rawRequestStr) {
        
        RawRequest rawRequest = new RawRequest(rawRequestStr);

        return new HttpRequest(
                RequestLineParser.parse(rawRequest.getRequestLine()),
                new RequestHeader(rawRequest.getHeaders()),
                new RequestBody(rawRequest.getBody())
        );
    }
```

Reply:
Parser 에 존재하는 로직을 HttpRequest 에 parse 함수를 만들어서 처리하라는 말씀이신거죠??❔

Reply:
네 맞아요
rawRequestStr 을 입력받아 parsing 하여 HttpRequest 를 생성하는 역할은 HttpRequest 의 정적 팩터리 메서드에 잘 어울릴 것 같아요

아래의 내용을 참고해보세요 : )
https://docs.oracle.com/javase/tutorial/datetime/overview/naming.html